app/models/comment.rb

```diff
@@ -46,14 +48,23 @@ class Comment < ActiveRecord::Base
     participation.unparticipate! if participation.present?
   end
 
-  def message
-    @message ||= Diaspora::MessageRenderer.new text
-  end
-
   def text= text
      self[:text] = text.to_s.strip #to_s if for nil, for whatever reason
   end
 
+  def people_allowed_to_be_mentioned
+    return @people_allowed_to_be_mentioned unless @people_allowed_to_be_mentioned.nil?
+    contacts = author.owner.try(:contact_people) if parent.public?
+
+    @people_allowed_to_be_mentioned = [*parent.participants, *contacts, parent.author].uniq
+  end
+
+  def subscribers
+    super.tap {|subscribers|
+      subscribers.concat(legitimate_mention_recipients) if parent.author.local?
```

Comment:
This doesn't work for relayed comments on public posts, where the comment-author is allowed to mention his contacts. The parent author would filter this persons and don't deliver the comment to them if they are not also in the contacts of the parent-author or have already participated.

Maybe we should just allow all people to be mentioned on public comments (like on public posts)? We can decide to send no notification for the mentioned person, if he doesn't have the information, that the author is sharing with him, if that is a problem, but we also don't do that for posts.